comp/core/agenttelemetry/agenttelemetryimpl/utils.go

```diff
@@ -0,0 +1,161 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// ---------------------------------------------------
+//
+// This is experimental code and is subject to change.
+//
+// ---------------------------------------------------
+
+package agenttelemetryimpl
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/itchyny/gojq"
+	dto "github.com/prometheus/client_model/go"
+)
+
+type jBuilder struct {
+	jqSource    *gojq.Query
```

Comment:
lThe idea of running JQ directly sounds good. It has a penalty on our side that we need to bring a 3rd party dependency into the mix. 

Is there any other way to achieve the same logic without adding support for JQ?

Reply:
We already use this module (https://github.com/DataDog/datadog-agent/blob/main/pkg/util/jsonquery/jsonquery.go#L13), which is further indirectly used in https://github.com/DataDog/datadog-agent/blob/main/pkg/compliance/resolver.go#L384 and https://github.com/DataDog/datadog-agent/blob/main/cmd/agent/common/autodiscovery.go#L319. Note to myself. Switch to our own wrapper - jsonquery

Reply:
Thanks for the context.

If I understand we use JQ to be able to modify the metrcis we emit juts by modifying the configuration, which would mean that we are able to change things faster rather than deploying the agent.

I still think that having this feature been use by customer (Still experimental) and being this dynamic might lead to unwanted consequences. Maybe a customer runs a complex JQ query (performance) Or creates an Invalid JQ  (shoot themself in the foot). Hard to migrate from using JQ (Once customers use it it might be difficult to remove t)

I'm raising this concerns, because we might reconsider shipping the initial implementation without JQ and see how adoption is and validate if we need this functionality. 

For now we could remove the `extra` feature:
```
extra:
        runner.workers.count: '.runnerStats.Workers.Count'
        runner.workers.utilization: '[.runnerStats.Workers | .. | objects | .Utilization] | add'
```

Extract those values manually for the `basic` template.

Reply:
discussed in OOTB